packages/@aws-cdk/aws-events/lib/matcher.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Event Pattern matcher.
+ */
+export abstract class Matcher {
```

Comment:
The other option instead of introducing dedicated factory class this functionality could become part of `EventPattern` type.